### PR TITLE
lint: "JavaScript" capitalization

### LINKED ISSUE #15755: Capitalization in text stings - "JavaScript"
**Is your feature request related to a problem? Please describe.**
JavaScript has 3 different capitalizations in different text strings. 

1. Correct mixed-case `JavaScript` (Key: user_actions_summary)
2. Incorrect lowercase `javascript` (Key: not_valid_js_addon)
3. Incorrect "title" case `Javascript` (Key: html_javascript_debugging)

Sources: While sometimes impeachable, https://en.wikipedia.org/wiki/JavaScript seems as good a compendium of references as any. https://www.w3.org/ is in accord. 

**Describe the solution you'd like**
For them all to be the same -- preferrably `JavaScript`.

**Describe alternatives you've considered**
Not complaining about it, because it is petty, minor, and causes no real problems. 

**Drawbacks to the proposed solution**
- Many translators will have been faithful to the capitalization of the original text, because (1) that's their job as translators, or (2) they were unaware of the inconsistency. 
- It's unknown which languages don't follow English proper noun capitalization, so the translated strings may need to be left undisturbed. 

**Additional context**
image

### PR DESCRIPTION
**Pull request by david-allison**
## Purpose / Description
not "javascript" or "Javascript"

## Fixes
* Fixes #15755

## Approach
Lint

## How Has This Been Tested?
Unit tests

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by BrayanDSO — APPROVED** [review]

File: lint-rules/src/main/java/com/ichi2/anki/lint/rules/TranslationTypo.kt [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import com.google.common.annotations.VisibleForTesting
+import com.ichi2.anki.lint.utils.Constants
+import org.w3c.dom.Element
+
+/**
+ * Ensures that strings are cased correctly
+ *
+ * `JavaScript`, not `Javascript`
+ */
+class TranslationTypo : ResourceXmlDetector(), XmlScanner {
```
Comment (lukstbit) [inline lint-rules/src/main/java/com/ichi2/anki/lint/rules/TranslationTypo.kt:32]: Naming complaint: _TranslationTypo_ is too vague, this issue specifically looks for proper case for _JavaScript_. Maybe _JavaScriptStringsCase_(or anything that mentions the search for JavaScript)?

**Review by lukstbit — APPROVED** [review]
Changes LGTM so consider this an approval.
In my opinion it feels kind of silly to introduce a full lint rule for this. I don't think there will be so many usages of _JavaScript_ to justify the extra lint run time(one usage now?). I would prefer a comment on the actual strings that include the word JavaScript(ex: _comment="Proper name is JavaScript"_) as context for translators.

**Comment by BrayanDSO** [comment]
> I would prefer a comment on the actual strings that include the word JavaScript(ex: comment="Proper name is JavaScript")

that's a good idea. And cutting linting time is always nice

File: lint-rules/src/main/java/com/ichi2/anki/lint/rules/TranslationTypo.kt (hunk 1 above)
Comment (david-allison) [reply lint-rules/src/main/java/com/ichi2/anki/lint/rules/TranslationTypo.kt:32]: I'll want to extend this class for other typography-related issues so we're not reading the whole XML tree too many times. Happy for another more generic suggestion

**Comment by david-allison** [comment]
My issue here is that we've let 'Javascript/javascript' get through reviews twice already in English. This has meant duplicated work for our translators. Which is especially bad as it trashes work from our less-regular translators over what amounts to a typo, reverting strings back to English

Realistically, we're not manually going to check the work of our translators, and I feel there's value in linting this at the 'sync translations' stage, if nothing else.

We're moving to addons, which will require more usages of JavaScript, 

Lint is run in parallel and does not dominate the runtime, so at worst case, adding a second won't impact time to merge

----

I'm not dead-set on this going in, at the end of the day it's a small cleanup, but I feel it'll have a positive long-term impact

**Comment by lukstbit** [comment]
Ok with me as well.

**Pull request merged**